### PR TITLE
 Let integration tests isolate generated api jars per intTestImage

### PR DESCRIPTION
**Pull request by bamboo**
See gradle/gradle-private#1467

### CODE REVIEW

File: subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/executer/AbstractGradleExecuter.java [hunk 1]
```diff
@@ -954,6 +955,8 @@ private boolean isSettingsFileAvailable() {
 
         if (useOwnUserHomeServices || (gradleUserHomeDir != null && !gradleUserHomeDir.equals(buildContext.getGradleUserHomeDir()))) {
             properties.put(REUSE_USER_HOME_SERVICES, "false");
+        } else {
```
Comment (oehme) [inline subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/executer/AbstractGradleExecuter.java]: ❌ This looks wrong to me, setting the override for all tests that **don't** have their own user home dir. It should be the opposite.

Comment (bamboo) [reply subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/executer/AbstractGradleExecuter.java]: Why the opposite? If a test already has their own user home dir, its API jar cache is already private.

Comment (oehme) [reply subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/executer/AbstractGradleExecuter.java]: Sorry, I got confused here. 

Another point though: Are you sure it's correct to skip this if `useOwnUserHomeServices` is true? I think it should have its own if-statement instead.

Comment (eskatos) [inline subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/executer/AbstractGradleExecuter.java]: :x: `useOwnUserHomeServices` is orthogonal, it should be complied with in all cases
from https://github.com/gradle/gradle/pull/6673/files#r215973768

**Pull request merged**